src/theory/sets/theory_sets_private.cpp

```diff
@@ -1358,7 +1359,7 @@ TrustNode TheorySetsPrivate::expandIsSingletonOperator(const Node& node)
 
   TypeNode setType = set.getType();
   Node boundVar = nm->mkBoundVar(setType.getSetElementType());
-  Node singleton = nm->mkNode(kind::SINGLETON, boundVar);
+  Node singleton = nm->mkSingleton(setType.getSetElementType(), boundVar);
```

Comment:
This is safe because we are extracting the element type from the set that is compared with the constructed singleton. `boundVar.getType()` is safe too.